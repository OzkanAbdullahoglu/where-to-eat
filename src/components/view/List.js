/* eslint-disable linebreak-style */
import React from 'react';
import PropTypes from 'prop-types';

import ListItem from './ListItem';

const List = ({
  fetchedData,
}) => (
  <div
    className="cards-layout cards-scrollbox scrollable-y cards-layout-flex-vertical"
    tabIndex="-1"
  >
    {fetchedData.map((item) => (
      <ListItem
        key={item.id}
        name={item.name}
        itemRate={item.rating}
        reviewCounts={item.review_count}
        itemCost={item.price}
        itemAddress={`${item.location.address1} ${item.location.address1} ${item.location.city}`}
        itemHours={item.hours}
        imageURL={item.photos}
        yelpURL={item.url}
        itemCategories={item.categories}
      />
    ))}
  </div>

);

List.propTypes = {
  fetchedData: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.array,
  ]),
};

export default List;
