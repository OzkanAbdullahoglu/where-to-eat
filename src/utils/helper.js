
export const openHoursManage = (obj) => {
  let willOpen = '';
  let willClosed = '';
  let openUntil = '';
  const date = new Date();
  const today = date.getDay();
  const time = parseInt(
    `${JSON.stringify(date.getHours())}${JSON.stringify(date.getMinutes())}`, 10
  );

  if (obj === undefined || obj.open[today] === undefined) {
    return '';
  }
  const closeTimeInt = parseInt(obj.open[today].end, 10);
  const openTime = `${obj.open[today].start.slice(0, 2)}:${obj.open[today].start.slice(2)}`;
  const closeTime = `${obj.open[today].end.slice(0, 2)}:${obj.open[
    today
  ].end.slice(2)}`;
  if (closeTimeInt - time < 100) {
    willClosed = `Close soon ${closeTime}`;
  } else if (time > closeTimeInt || time < closeTimeInt) {
    willOpen = `Opens at ${openTime}`;
  } else {
    openUntil = `Open until ${closeTime}`;
  }
  return (
    willClosed || willOpen || openUntil
  );
};

export const getCategoriesTitle = (arr) => {
  const titleArr = arr.map((obj) => obj.title);
  const concatenateTitles = titleArr.reduce(
    (accTitles, singleTitle) => `${accTitles}, ${singleTitle}`
  );
  return concatenateTitles;
};
