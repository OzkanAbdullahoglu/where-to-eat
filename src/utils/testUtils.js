/* eslint-disable linebreak-style */
import { createStore, applyMiddleware } from 'redux';
import { middlewares } from '../store';
import rootReducer from '../reducers';


export const storeFactory = (initalState) => {
  const createStoreWithMiddleware = applyMiddleware(...middlewares)(createStore);
  return createStoreWithMiddleware(rootReducer, initalState);
}
;
