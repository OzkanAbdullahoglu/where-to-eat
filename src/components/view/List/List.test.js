/* eslint-disable linebreak-style */
import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import List from './List';
import { fetchedDataMock } from '../../../utils/testUtils';

describe('List Component', () => {
  const props = {
    fetchedData: fetchedDataMock,
  };
  const listWrapper = shallow(<List {...props} />);
  it('renders List Component', () => {
    expect(toJson(listWrapper)).toMatchSnapshot();
  });
  it('contains a List Item Component', () => {
    expect(listWrapper.find('ListItem').exists()).toBe(true);
  });

  describe('List Component Passing Props Correctly', () => {
    it('passing `imageURL` props', () => {
      const imageURLProp = listWrapper.find('ListItem').prop('imageURL');
      expect(imageURLProp).toEqual(props.fetchedData[0].photos);
    });
    it('passing `key` props', () => {
      const keyProp = listWrapper.find('ListItem').prop('keyTest');
      expect(keyProp).toEqual(props.fetchedData[0].id);
    });
    it('passing `name` props', () => {
      const nameProp = listWrapper.find('ListItem').prop('name');
      expect(nameProp).toEqual(props.fetchedData[0].name);
    });
    it('passing `itemRate` props', () => {
      const itemRateProp = listWrapper.find('ListItem').prop('itemRate');
      expect(itemRateProp).toEqual(props.fetchedData[0].rating);
    });
    it('passing `reviewCounts` props', () => {
      const reviewCountsProp = listWrapper.find('ListItem').prop('reviewCounts');
      expect(reviewCountsProp).toEqual(props.fetchedData[0].review_count);
    });
    it('passing `itemCost` props', () => {
      const itemCostProp = listWrapper.find('ListItem').prop('itemCost');
      expect(itemCostProp).toEqual(props.fetchedData[0].price);
    });
    it('passing `itemAddress` props', () => {
      const itemAddressProp = listWrapper.find('ListItem').prop('itemAddress');
      expect(itemAddressProp).toEqual(`${props.fetchedData[0].location.address1} ${props.fetchedData[0].location.address1} ${props.fetchedData[0].location.city}`);
    });
    it('passing `itemHours` props', () => {
      const itemHoursProp = listWrapper.find('ListItem').prop('itemHours');
      expect(itemHoursProp).toEqual(props.fetchedData[0].hours);
    });
    it('passing `yelpURL` props', () => {
      const yelpURLProp = listWrapper.find('ListItem').prop('yelpURL');
      expect(yelpURLProp).toEqual(props.fetchedData[0].url);
    });
    it('passing `itemCategories` props', () => {
      const itemCategoriesProp = listWrapper.find('ListItem').prop('itemCategories');
      expect(itemCategoriesProp).toEqual(props.fetchedData[0].categories);
    });
  });
});
