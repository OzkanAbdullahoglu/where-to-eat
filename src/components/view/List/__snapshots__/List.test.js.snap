// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List Component renders List Component 1`] = `
<div
  className="cards-layout cards-scrollbox scrollable-y cards-layout-flex-vertical"
  tabIndex="-1"
>
  <ListItem
    imageURL={
      Array [
        "https://s3-media2.fl.yelpcdn.com/bphoto/BIHWiUEsQRukg-tccQ8H-w/o.jpg",
      ]
    }
    itemAddress="test address heading test address heading test address city"
    itemCategories={
      Array [
        Object {
          "title": "Turkish",
        },
      ]
    }
    itemCost="$$$$$"
    itemHours={
      Array [
        Object {
          "open": Array [
            Object {
              "day": 0,
              "end": "2200",
              "is_overnight": false,
              "start": "1200",
            },
            Object {
              "day": 1,
              "end": "2200",
              "is_overnight": false,
              "start": "1200",
            },
            Object {
              "day": 2,
              "end": "2200",
              "is_overnight": false,
              "start": "1200",
            },
            Object {
              "day": 3,
              "end": "2200",
              "is_overnight": false,
              "start": "1200",
            },
            Object {
              "day": 4,
              "end": "2200",
              "is_overnight": false,
              "start": "1200",
            },
            Object {
              "day": 5,
              "end": "2200",
              "is_overnight": false,
              "start": "1200",
            },
            Object {
              "day": 6,
              "end": "2200",
              "is_overnight": false,
              "start": "1200",
            },
          ],
        },
      ]
    }
    itemRate={4}
    key="123456789test"
    keyTest="123456789test"
    name="test name"
    reviewCounts={123}
    yelpURL="https://www.yelp.com/biz/uluda%C4%9F-kebap%C3%A7%C4%B1s%C4%B1-cemal-ve-cemil-usta-istanbul?adjust_creative=lQpxUARWc1tWB0NFQoQQZQ&utm_campaign=yelp_api_v3&utm_medium=api_v3_graphql&utm_source=lQpxUARWc1tWB0NFQoQQZQ"
  />
</div>
`;
